File: codeforces/virtual/310/2.java
/* Author: Hemanth Kumar Veeranki */
/* Handle: harry7 */
import java.io.*;
import java.util.*;
import java.math.*;
public class Main{
	public static void main(String[] args){
		Scanner sc=new Scanner(System.in); /*sc for taking input*/
		PrintStream op=System.out;/*op for printing output*/
		int n,i;
		n=sc.nextInt();
		int[] a=new int[n];
		for(i=0;i<n;i++)a[i]=sc.nextInt();
		while(a[0]!=0){
		a[0]=(a[0]+1)%n;
		for(i=1;i<n;i++)
			if(i%2==1)a[i]=(a[i]+n-1)%n;
			else a[i]=(a[i]+1)%n;
		}
		for(i=0;i<n;i++)if(a[i]!=i){op.print("No");return ;}
		op.print("Yes");
	}
}

File: hackerrank/algorithms/contests/101hackjun2015/2.py
""" 
    Author: Hemanth Kumar Veeranki
    Handle:harry7

"""
from math import *
def gcd(a,b):
    if(a==0):return b
    return gcd(b%a,a)
for _ in  xrange(input()):
    a,b=map(int,raw_input().split())
    r=a
    if(a>b):a%=b
    p=b/4.0-a
    if(p<0):p+=b
    r+=p
    r=int(floor(r))
    p=gcd(r,b)
    if(r%b==0):print r,"0/1"
    else: 
        a=r%b
        print r,str(a/p)+"/"+str(b/p)
